code/datums/status_effects/debuffs.dm

```diff
@@ -67,8 +67,19 @@
 	return ..()
 
 /datum/status_effect/incapacitating/sleeping/tick()
-	if(owner.getStaminaLoss())
-		owner.adjustStaminaLoss(-0.5) //reduce stamina loss by 0.5 per tick, 10 per 2 seconds
+	if(owner.maxHealth != 0)
+		var/health_ratio = owner.health / owner.maxhealth
+		if(health_ratio > 0.8)
+			var/healing = -0.2
+			if((locate(/obj/structure/bed) in owner.loc))
+				healing -= 0.3
+			else
+				if((locate(/obj/structure/table) in owner.loc))
```

Comment:
Doesn't this code only detect if they are on the same tile as the bed, not if they actually buckled to it?

Reply:
correct

Reply:
also it is maxHealth not maxhealth you sausage

Reply:
if you sleep you lay on the bed anyways

Reply:
This means an unlimited number of people can use the same bed to heal, which is very silly.

Reply:
one bomb to ruin them all is a feature...
*leans in* and so is erp

Reply:
Yes actually leave that in please.